### PR TITLE
GS on personal A/B: Tootips to reflect the A/B plan.

### PR DESCRIPTION
**Pull request by rcrdortiz**
Related to https://github.com/Automattic/dotcom-forge/issues/2859

## Proposed Changes

This PR adjusts the message displayed in the `ThemeTypeBadgeTooltip` component based on a site's global styles status. The tooltip message differentiates between sites that have global styles in their Personal plan, and those that require a Premium plan upgrade to unlock this feature. We achieve this by utilizing the `useSiteGlobalStylesStatus` hook to fetch the site's global styles status and then adjusting the tooltip message accordingly.

The scope of these changes is limited to A/B testing GS on personal.

| Before | After |
------|------
Captura de Pantalla 2023-07-04 a las 11 17 05|Captura de Pantalla 2023-07-04 a las 11 18 01

## Testing Instructions

1. Open the Calypso live link.
2. Make sure your site is on a `FREE` plan.
3. Navigate to the themes section either in the administration section `/themes/{yoursite}` or the site creation flow `/setup/site-setup/designSetup?siteSlug={yoursite}&ref=site-selector`
4. Select a style variation from a free theme that has style variations
5. If your site has global styles included in the Personal plan, the tooltip should read, "Unlock this style, and tons of other features, by upgrading to a Personal plan."
6. If your site doesn't have global styles included in the Personal plan, the tooltip should read, "Unlock this style, and tons of other features, by upgrading to a Premium plan."

You can change your group here (21268-explat-experiment), you also need to clear the cache when switching between groups by executing this command in your sandbox:
`wp_cache_delete( 'global-styles-on-personal-{siteId}', 'a8c_experiments' );` 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by dsas — APPROVED** [review]
both showcase and design picker lgtm 👍

**Review by dsas — CHANGES_REQUESTED** [review]
Test failures look relevant though, I triggered a rebuild and they recurred.

**Comment by Copons** [comment]
It seems the [tests](https://github.com/Automattic/wp-calypso/blob/b6b5b0ae887c23dc107399edc1895add0b5abb96/client/components/theme-type-badge/test/index.jsx) are failing here:

https://github.com/Automattic/wp-calypso/blob/b6b5b0ae887c23dc107399edc1895add0b5abb96/client/state/sites/hooks/use-site-global-styles-status.ts#L89-L95

because of `No QueryClient set, use QueryClientProvider to set one`.

The change was introduced in https://github.com/Automattic/wp-calypso/pull/78895, which (I presume) did not trigger anything because it was not yet wired with unit-tested components.

---

Tests are also failing because three separate `screen.queryByTestId( 'upsell-header' )` turn out to be `null`.
They then break at `expect( screen.queryByTestId( 'upsell-header' ).innerHTML ).toBe( ... )`.

Not sure if this is a side effect caused by the QueryClient error, though.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8350873

Thank you @rcrdortiz for including a screenshot in the description! This is really helpful for our translators.

File: client/components/theme-type-badge/test/index.jsx [hunk 1]
```diff
@@ -43,8 +44,12 @@ describe( 'ThemeTypeBadge', () => {
 
 	describe( 'Premium theme popover', () => {
 		test( 'Free site', async () => {
+			const queryClient = new QueryClient();
+
 			const { container } = renderWithState(
-				<ThemeTypeBadge siteId={ 123 } siteSlug="example.com" themeId="premium/test" />
+				<QueryClientProvider client={ queryClient }>
```
Comment (mmtr) [inline client/components/theme-type-badge/test/index.jsx]: I think it'd be easier if you just mock `useSiteGlobalStylesStatus`

Comment (mmtr) [reply client/components/theme-type-badge/test/index.jsx]: See example: https://github.com/Automattic/wp-calypso/blob/1f55d9f821f39b44ad46988a0e97f1fd44eb7cdd/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/test/unified-design-picker.tsx#L38-L40

Comment (rcrdortiz) [reply client/components/theme-type-badge/test/index.jsx]: Changed

File: client/components/theme-type-badge/test/index.jsx [hunk 2]
```diff
@@ -7,6 +7,13 @@ import { Provider } from 'react-redux';
 import configureStore from 'redux-mock-store';
 import ThemeTypeBadge from '../';
 
+jest.mock( 'calypso/state/sites/hooks/use-site-global-styles-status', () => ( {
+	useSiteGlobalStylesStatus: () => ( {
+		shouldLimitGlobalStyles: false,
+		globalStylesInPersonalPlan: false,
+	} ),
```
Comment (mmtr) [inline client/components/theme-type-badge/test/index.jsx]: Nitpick: `shouldLimitGlobalStyles` probably doesn't need to be mocked
```suggestion
	useSiteGlobalStylesStatus: () => ( {
		globalStylesInPersonalPlan: false,
	} ),
```

**Review by mmtr — APPROVED** [review]
Tests well!

**Review by dsas — APPROVED** [review]

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8350873).

**Pull request merged**